File: mobile/src/__tests__/entities/family/carePushCopy.test.ts
import { buildCarePushCopy } from '@/entities/family/lib/carePushCopy';

describe('buildCarePushCopy', () => {
  it('taken — 닉네임 포함 안부 톤', () => {
    const copy = buildCarePushCopy('taken', '엄마');
    expect(copy.title).toBeTruthy();
    expect(copy.body).toContain('엄마');
    expect(copy.body).not.toMatch(/점수|감시|CCTV/i);
  });

  it('stuck — 닉네임 없으면 fallback', () => {
    const copy = buildCarePushCopy('stuck_escalate', null);
    expect(copy.title).toBeTruthy();
    expect(copy.body.length).toBeGreaterThan(0);
  });
});

File: android-app/core/src/test/kotlin/com/jinlabs/yakok/core/family/CarePushCopyTest.kt
package com.jinlabs.yakok.core.family

import com.jinlabs.yakok.core.constants.CarePush
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import org.junit.jupiter.api.Test

class CarePushCopyTest {
    @Test
    fun taken_includesNickname() {
        val copy = CarePushCopy.build(CarePushKind.Taken, "엄마")
        assertEquals("약 챙겼어요", copy.title)
        assertEquals("엄마 님이 약을 먹었어요", copy.body)
        assertFalse(copy.body.contains("점수"))
        assertFalse(copy.body.contains("감시"))
    }

    @Test
    fun stuck_fallbackNickname() {
        val copy = CarePushCopy.build(CarePushKind.StuckEscalate, "  ")
        assertEquals("안부가 궁금해요", copy.title)
        assertEquals("가족 님 복약 체크가 멈춘 것 같아요", copy.body)
    }

    @Test
    fun channelIds_matchRn() {
        assertEquals("care-taken", CarePushKind.Taken.channelId)
        assertEquals("care-stuck", CarePushKind.StuckEscalate.channelId)
        assertEquals("announcement", CarePush.ChannelAnnouncement)
    }
}
